File: 1773-count-items-matching-a-rule/1773-count-items-matching-a-rule.java
class Solution {
    public int countMatches(List<List<String>> items, String ruleKey, String ruleValue) {
        int i=-1;
        switch(ruleKey){
            case "type":
                i=0;
                break;
            case "color":
                i=1;
                break;
            default:
                i=2;
            
        }
        int count=0;
        // int k=i;
        for (List<String> j : items) {
            if(j.get(i).equals(ruleValue)){
                count++;
            }
        }
        return count;
        
    }
}

File: 1773-count-items-matching-a-rule/1773-count-items-matching-a-rule.js
/**
 * @param {string[][]} items
 * @param {string} ruleKey
 * @param {string} ruleValue
 * @return {number}
 */
var countMatches = function(items, ruleKey, ruleValue) {
  const rules={
    'type':0,
    'color':1,
    'name':2
  }
  // let count=0
  // console.log(value)
   let match= items.map(e=>e).reduce((acc,cur)=>cur[rules[ruleKey]]===ruleValue?acc+1:acc,0)
  return match
};